openedx/core/djangoapps/discussions/models.py

```diff
@@ -0,0 +1,61 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from jsonfield import JSONField
+from model_utils.models import TimeStampedModel
+from opaque_keys.edx.django.models import LearningContextKeyField
+from simple_history.models import HistoricalRecords
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class LearningContextDiscussionConfig(TimeStampedModel):
+    """
+    Associates a learning context with discussion provider and configuration
+    """
+
+    context_key = LearningContextKeyField(
+        primary_key=True,
+        db_index=True,
+        unique=True,
+        max_length=255,
+        verbose_name=_("Learning Context"),
+    )
+    enabled = models.BooleanField(
+        default=True,
+        help_text=_("If disabled, the discussions in the associated learning context/course will be disabled.")
+    )
+    lti_configuration = JSONField(
+        blank=True,
+        default={},
+        help_text=_("The LTI configuration data for this context/provider."),
+        # - launch_url
+        # - api_key
+        # - api_secret
+    )
+    plugin_configuration = JSONField(
+        blank=True,
+        default={},
+        help_text=_("The plugin configuration data for this context/provider."),
+    )
+    provider = models.CharField(
+        blank=False,
+        db_index=False,
+        max_length=100,
+        verbose_name=_("Discussion provider"),
+        help_text=_("The discussion tool/provider."),
```

Comment:
Maybe `The discussion tool/provider's id.`?

Reply:
Renamed to `provider_type`.